### PR TITLE
Fix gosec issues: fix unhandled errors

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:
This PR address issues find by gosec. The issue is that we ignore few errors. 
Defaulting to log an error can give us visibility if they occur and therefore this approach is used.

**Special notes for your reviewer**:
Waiting on https://github.com/kubevirt/kubevirt/pull/4504
**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-launcher/virtwrap/util/libvirt_helper.go [hunk 1]
```diff
@@ -266,7 +270,7 @@ func StartVirtlog(stopChan chan struct{}, domainName string) {
 					log.Log.Reason(err).Error("failed to catch virtlogd logs")
 					return
 				}
-				defer util.CloseIOAndCheckErr(file, nil)
+				defer file.Close()
```
Comment (dhiller) [inline pkg/virt-launcher/virtwrap/util/libvirt_helper.go:273]: Is that intended?

File: pkg/virt-operator/creation/components/secrets.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ import (
 	"kubevirt.io/kubevirt/pkg/certificates/triple/cert"
 )
 
+// #nosec
```
Comment (dhiller) [inline pkg/virt-operator/creation/components/secrets.go:23]: I'd expect an explanation why this is a false positive or not an issue.

File: pkg/virt-operator/util/config.go [hunk 3]
```diff
@@ -76,7 +76,7 @@ const (
 	operatorImageRegex = "^(.*)/(.*)virt-operator([@:].*)?$"
 
 	// Prefix for env vars that will be passed along
-	PassthroughEnvPrefix = "KV_IO_EXTRA_ENV_"
+	PassthroughEnvPrefix = "KV_IO_EXTRA_ENV_" // #nosec
```
Comment (dhiller) [inline pkg/virt-operator/util/config.go:79]: I'd expect an explanation why this is not an issue.

**Review by dhiller — COMMENTED** [review]
Looks good in general, some questions inline.

File: pkg/virtctl/vnc/vnc.go [hunk 4]
```diff
@@ -145,7 +145,8 @@ func (o *VNC) Run(cmd *cobra.Command, args []string) error {
 		// Don't set deadline if only proxy is running and VNC is to be connected manually
 		if !proxyOnly {
 			// exit early if spawning vnc client fails
-			ln.SetDeadline(time.Now().Add(LISTEN_TIMEOUT))
+			err := ln.SetDeadline(time.Now().Add(LISTEN_TIMEOUT))
+			listenResChan <- fmt.Errorf("Failed to set deadline on listener. %v", err)
```
Comment (ezrasilvera) [inline pkg/virtctl/vnc/vnc.go:149]: Isn't this dependent on err != nil ? or something

File: pkg/virtctl/templates/templates.go [hunk 5]
```diff
@@ -52,7 +52,9 @@ func ExactArgs(nameOfCommand string, n int) cobra.PositionalArgs {
 	return func(cmd *cobra.Command, args []string) error {
 		if len(args) != n {
 			fmt.Printf("fatal: Number of input parameters is incorrect, %s accepts %d arg(s), received %d\n\n", nameOfCommand, n, len(args))
-			cmd.Help()
+			if err := cmd.Help(); err != nil {
+				panic("Help doesn't return error")
```
Comment (ezrasilvera) [inline pkg/virtctl/templates/templates.go:56]: 1) The error message is not clear here.  
2) Why panic and not returning error ?

File: pkg/virt-operator/util/config.go [hunk 6]
```diff
@@ -431,7 +431,9 @@ func (c *KubeVirtDeploymentConfig) generateInstallStrategyID() {
 	// reason: sha1 is not used for encryption but for creating a hash value
 	hasher := sha1.New()
 	values := getStringFromFields(*c)
-	hasher.Write([]byte(values))
+	if _, err := hasher.Write([]byte(values)); err != nil {
+		panic("Error with sha1 is not possible")
```
Comment (ezrasilvera) [inline pkg/virt-operator/util/config.go:435]: Are we sure we can just panic here ?  Maybe we need to change the function signature, return and error , and change the caller to check the error and return the proper values further (error, nil, etc..)

File: pkg/virt-operator/resource/generate/install/strategy.go [hunk 7]
```diff
@@ -244,67 +249,112 @@ func DumpInstallStrategyToConfigMap(clientset kubecli.KubevirtClient, operatorNa
...
+		}
 	}
 	for _, entry := range strategy.configMaps {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
+	}
+	if err := writer.Flush(); err != nil {
+		panic(fmt.Errorf("Flush on bytes should be safe, %v", err))
```
Comment (ezrasilvera) [inline pkg/virt-operator/resource/generate/install/strategy.go:354]: Again - why panic and not returning error ?

File: pkg/virt-operator/resource/generate/install/strategy.go [hunk 8]
```diff
@@ -244,67 +249,112 @@ func DumpInstallStrategyToConfigMap(clientset kubecli.KubevirtClient, operatorNa
...
+			return nil, err
+		}
 	}
 	for _, entry := range strategy.roleBindings {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
 	}
+	delimeter := []byte("---\n")
```
Comment (ezrasilvera) [inline pkg/virt-operator/resource/generate/install/strategy.go:289]: Why do we need the variable?  It seems it's used once

**Comment by ezrasilvera** [comment]
@xpivarc This PR introduces many new `panic` statements. As panic change dramatically the behavior (as oppose to returning an error) , just need to verify that on all the new panic you indeed want to abort completely .

File: pkg/virtctl/vnc/vnc.go (hunk 4 above)
Comment (xpivarc) [reply pkg/virtctl/vnc/vnc.go:149]: You are right

File: pkg/virtctl/templates/templates.go (hunk 5 above)
Comment (xpivarc) [reply pkg/virtctl/templates/templates.go:56]: I probably didn't find the return statement in the Help() function, I will append this error to one which is below.

File: pkg/virt-operator/util/config.go (hunk 6 above)
Comment (xpivarc) [reply pkg/virt-operator/util/config.go:435]: We can add the return but I don't think that error is possible here. (I am not saying we should keep the panic, I actually used `panic` so we can discuss the places which I couldn't decide what to do)

File: pkg/virt-operator/resource/generate/install/strategy.go (hunk 7 above)
Comment (xpivarc) [reply pkg/virt-operator/resource/generate/install/strategy.go:354]: I don't think that the error is possible but it's easy to return here.

**Pull request closed without merging**